cdap-app-templates/cdap-etl/cdap-data-pipeline/src/main/java/io/cdap/cdap/datapipeline/draft/DraftStoreRequest.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright © 2020 Cask Data, Inc.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ *
+ */
+
+package io.cdap.cdap.datapipeline.draft;
+
+import io.cdap.cdap.api.artifact.ArtifactSummary;
+
+import java.util.Objects;
+import javax.annotation.Nullable;
+
+/**
+ * Request to store a draft.
+ *
+ * @param <T> Type of config that this DraftRequest contains
+ */
+public class DraftStoreRequest<T> {
+  private final String previousHash;
+  private final String name;
+  private final String description;
+  private final int revision;
+  private final ArtifactSummary artifact;
+  private final T config;
+
+  public DraftStoreRequest(@Nullable T config, String previousHash, String name, String description, int revision,
+                           @Nullable ArtifactSummary artifact) {
+    this.config = config;
+    this.previousHash = previousHash;
+    this.name = name;
+    this.description = description;
+    this.revision = revision;
+    this.artifact = artifact;
+  }
+
+  public String getDescription() {
+    return description;
+  }
+
+  public String getName() {
+    return name == null ? "" : name;
+  }
+
+  public ArtifactSummary getArtifact() {
+    return artifact;
+  }
+
+  public T getConfig() {
```

Comment:
Annotate methods with `@Nullable` if null can be returned (config and artifact, I believe).

Reply:
done